### PR TITLE
Fix default design condenser inlet and design evaporator outlet temperatures for Chiller:ConstantCOP

### LINKED ISSUE #10157: Chiller:ConstantCOP default design Evaporator Out and Condenser In temperatures
Issue overview
--------------
Discovered during PR #10143 (ASHRAE Std 229 ruleset model description) development and reviewing process (thanks @rraustad), the default design Evaporator out and Condenser In temperatures are not set. These two variables are left with zero initialization since creation and then are not changed for the chiller constant cop modeling. See the discussion details [here](https://github.com/NREL/EnergyPlus/pull/10143#issuecomment-1685622647).

It makes sense that a constant COP model does not need input fields for the condenser in and evaporator out design temperatures. From the point of modeling need, this treatment does make perfect sense. However, it is suggested to leave these two design conditions to more realistic default values, such as 29.44C and 6.67C respectively (which are also used for default for some other chiller types, such as the ASHRAE Std 205 chiller models), rather than leaving them as zeros.

It will make a difference in the output after PR# 10143 is merged for the chillers table (so that's also the diffs expected). It should also worth a documentation about the issue in a separate post (current issue post).

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): v23.2 pre-release, and earlier versions
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (any example file with a Chiller:ConstantCOP object, such as CoolingTower.idf would do, if one is needed)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #10157
 - Adds more realistic default design condenser inlet and evaporator outlet temperatures for Chiller:ConstantCOP, by using 29.44C (85F) for design condenser inlet temperature and 6.67C (44F) for design evaporator outlet temperature, to replace the original zero (0.0C) default initializations.  
 - Tabular output diffs are expected for test files that contain Chiller:ConstantCOP objects.
 
**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/PlantChillers.cc [hunk 1]
```diff
@@ -6539,6 +6539,10 @@ namespace PlantChillers {
                                             state.dataIPShortCut->cAlphaArgs(8)));
                 }
             }
+
+            // set default design condenser in and evaporator out temperatures
+            thisChiller.TempDesCondIn = 29.44; // Degree Celsius, or 85 Degree Fahrenheit
+            thisChiller.TempDesEvapOut = 6.67; // Degree Celsius, or 44 Degree Fahrenheit
```
Comment (rraustad) [inline src/EnergyPlus/PlantChillers.cc:6545]: That should do it!

image

**Review by rraustad — APPROVED** [review]
This will fill in the blanks in the table.

File: src/EnergyPlus/PlantChillers.cc (hunk 1 above)
Comment (lymereJ) [reply src/EnergyPlus/PlantChillers.cc:6545]: @rraustad @jcyuan2020 - This appears to be from AHRI 550/590, the IP version of the standard. Curiously, AHRI 551/591, the SI version of the standard, has slightly different rated conditions (when you convert the temperature from deg. C to deg. F). Maybe there should be a note about it.

Here's same table but in 551/591 and for example check out the leaving temperature (7 deg. C, not quite 6.67 deg. F):

image

**Comment by rraustad** [comment]
I spoke too soon. Yes, water-cooled uses an entering temperature of 85F, but there are columns in the Standards Rating table for evaporatively-cooled and air-cooled that show 95F. So the default is actually dependent on condenser type.

    Chiller:ConstantCOP,
      A6 , \field Condenser Type
           \type choice
           \key AirCooled
           \key WaterCooled
           \key EvaporativelyCooled
           \default AirCooled

File: src/EnergyPlus/PlantChillers.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/PlantChillers.cc:6545]: SI versions are notoriously inaccurate due to numeric conversion (even though you see 2 decimals as 7.00). I think I would stay with the IP ratings, but that's just me. I saw the same thing with #10117.

**Review by rraustad — CHANGES_REQUESTED** [review]
Add the initialization based on condenser type.

**Comment by rraustad** [comment]
Example file AirCooledElectricChiller results using develop branch:

image

**Comment by rraustad** [comment]
This branch looks correct now.

    Chiller:ConstantCOP,
      Little Chiller,          !- Name
      25000,                   !- Nominal Capacity {W}
      2.5,                     !- Nominal COP {W/W}
      0.0011,                  !- Design Chilled Water Flow Rate {m3/s}
      0.0011,                  !- Design Condenser Water Flow Rate {m3/s}
      Little Chiller Inlet Node,  !- Chilled Water Inlet Node Name
      Little Chiller Outlet Node,  !- Chilled Water Outlet Node Name
      Little Chiller Condenser Inlet Node,  !- Condenser Inlet Node Name
      Little Chiller Condenser Outlet Node,  !- Condenser Outlet Node Name
      AirCooled,               !- Condenser Type
      LeavingSetpointModulated,!- Chiller Flow Mode
      ;                        !- Sizing Factor


image

**Comment by rraustad** [comment]
Unit tests run successfully.

**Review by rraustad — APPROVED** [review]
Code change is accurate and unit test covers all possible condenser types.

**Comment by rraustad** [comment]
I was curious why Design Size Reference Condenser Fluid Flow shows 0. This just verifies there is a value.

image

File: src/EnergyPlus/PlantChillers.cc [hunk 2]
```diff
@@ -6539,6 +6539,19 @@ namespace PlantChillers {
...
+            if (thisChiller.CondenserType == DataPlant::CondenserType::WaterCooled) {
+                thisChiller.TempDesCondIn = 29.44; // Degree Celsius, or 85 Degree Fahrenheit
+            } else if (thisChiller.CondenserType == DataPlant::CondenserType::AirCooled) {
+                thisChiller.TempDesCondIn = 35.0; // Degree Celsius, or 95 Degree Fahrenheit
+            } else if (thisChiller.CondenserType == DataPlant::CondenserType::EvapCooled) {
+                thisChiller.TempDesCondIn = 35.0; // Degree Celsius, or 95 Degree Fahrenheit
+            } else {
+                thisChiller.TempDesCondIn = 35.0; // Degree Celsius, or 95 Degree Fahrenheit
+            }
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantChillers.cc:6554]: In the future, these things based on enums can be switch blocks, which add a bit of readability (you aren't continually repeating the conditional), but this is not a big deal, and looks good.

File: tst/EnergyPlus/unit/ChillerConstantCOP.unit.cc [hunk 3]
```diff
@@ -163,3 +163,94 @@ TEST_F(EnergyPlusFixture, ChillerConstantCOP_WaterCooled_Autosize)
...
+
+    auto &thisChiller_2 = state->dataPlantChillers->ConstCOPChiller(2);
+
+    EXPECT_NEAR(thisChiller_2.TempDesCondIn, 35.0, 1e-3);
+    EXPECT_NEAR(thisChiller_2.TempDesEvapOut, 6.67, 1e-3);
+
+    auto &thisChiller_3 = state->dataPlantChillers->ConstCOPChiller(3);
+
+    EXPECT_NEAR(thisChiller_3.TempDesCondIn, 35.0, 1e-3);
+    EXPECT_NEAR(thisChiller_3.TempDesEvapOut, 6.67, 1e-3);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ChillerConstantCOP.unit.cc:255]: Nice!

**Review by Myoldmopar — APPROVED** [review]
LGTM I'll test it out a bit.

**Comment by Myoldmopar** [comment]
Decent is showing lots of table diffs, which I don't want to pollute the rest of the merge queue, so this is likely ready, but will be held briefly.  Thanks @jcyuan2020

**Comment by Myoldmopar** [comment]
OK, I think I'll do this next.  It's the last one on the current merge queue, so while it will cause diffs, I need to pivot to deeper reviews, and it's a good spot for this.  I'll run a quick check and then merge if happy.

**Comment by Myoldmopar** [comment]
All good here.  Merging this one.  This will cause table diffs in any more CI runs that are about to happen, but we will have to handle that at some point, why not now?  Thanks @jcyuan2020

**Pull request merged**